homeassistant/components/homewizard_energy/config_flow.py

```diff
@@ -0,0 +1,187 @@
+
+        _LOGGER.debug("config_flow async_step_zeroconf")
+
+        # Validate doscovery entry
+        if (
+            "host" not in discovery_info
+            or "api_enabled" not in discovery_info["properties"]
+            or "path" not in discovery_info["properties"]
+            or "product_name" not in discovery_info["properties"]
+            or "product_type" not in discovery_info["properties"]
+            or "serial" not in discovery_info["properties"]
+        ):
+            return self.async_abort(reason="invalid_discovery_parameters")
+
+        if (discovery_info["properties"]["path"]) != "/api/v1":
+            return self.async_abort(reason="unsupported_api_version")
+
+        if (discovery_info["properties"]["api_enabled"]) != "1":
+            return self.async_abort(reason="api_not_enabled")
+
+        # Pass parameters
+        entry_info = {
+            CONF_IP_ADDRESS: discovery_info["host"],
+            CONF_PORT: discovery_info["port"],
+        }
+
+        return await self.async_step_check(entry_info)
+
+    async def async_step_check(self, entry_info):
+        """Validate API connection and fetch metadata."""
+
+        _LOGGER.debug("config_flow async_step_check")
+
+        # Make connection with device
+        energy_api = aiohwenergy.HomeWizardEnergy(entry_info[CONF_IP_ADDRESS])
+
+        initialized = False
+        try:
+            with async_timeout.timeout(10):
+                await energy_api.initialize()
+                if energy_api.device is not None:
+                    initialized = True
+
+        except aiohwenergy.DisabledError:
+            _LOGGER.error("API disabled, API must be enabled in the app")
+            return self.async_abort(reason="api_not_enabled")
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.error(
+                "Error connecting with Energy Device at %s",
+                entry_info[CONF_IP_ADDRESS],
+            )
+            return self.async_abort(reason="unknown_error")
+
+        finally:
+            await energy_api.close()
+
+        if not initialized:
+            _LOGGER.error("Initialization failed")
+            return self.async_abort(reason="unknown_error")
+
+        # Validate metadata
+        if energy_api.device.api_version != "v1":
+            return self.async_abort(reason="unsupported_api_version")
+
+        if energy_api.device.product_type not in SUPPORTED_DEVICES:
+            _LOGGER.error(
+                "Device (%s) not supported by integration",
+                energy_api.device.product_type,
+            )
+            return self.async_abort(reason="device_not_supported")
+
+        # Configure device
+        entry_info["product_name"] = energy_api.device.product_name
+        entry_info["product_type"] = energy_api.device.product_type
+        entry_info["serial"] = energy_api.device.serial
+
+        self.context[CONF_HOST] = entry_info[CONF_IP_ADDRESS]
```

Comment:
Don't store this in context but just on this config flow instance

Reply:
Removed it with 79f574f2d05a226bddeb6ec3016f667565132e72 and https://github.com/home-assistant/core/pull/55812/commits/edac621182d6f9cbf1d45c389e453d479395844f